modules/isd/src/ResidueProteinProximityRestraint.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ *  \file isd/ResidueProteinProximityRestraint.h
+ *  \brief Restraint a selection of particles (eg. a residue or 
+ *  segment) to be within  a certain distance of a second 
+ *  selection of particles (eg. a protein). 
+ *  Use to model data from mutagenesis experiments that disrupt
+ *  protein-protein interactions.
+ *
+ *  Copyright 2007-2022 IMP Inventors. All rights reserved.
+ *
+ */
+
+#include <IMP/isd/ResidueProteinProximityRestraint.h>
+#include <IMP/algebra/VectorD.h>
+#include <IMP/core/XYZ.h>
+#include <IMP/isd/Scale.h>
+#include <IMP/container/ListPairContainer.h>
+#include <boost/math/special_functions/erf.hpp>
+#include <boost/unordered_map.hpp>
+#include <boost/tuple/tuple.hpp>
+#include <math.h>
+
+IMPISD_BEGIN_NAMESPACE
+
+ResidueProteinProximityRestraint::ResidueProteinProximityRestraint(Model* m,
+						 Float cutoff,
+						 Float sigma,
+                                                 Float xi,
+						 bool part_of_log_score,
+						 std::string name):
+  Restraint(m,name),
+  cutoff_(cutoff),
+  sigma_(sigma),
+  xi_(xi),
+  part_of_log_score_(part_of_log_score)
+  {
+}
+
+// add a contribution: general case
+void ResidueProteinProximityRestraint::add_pairs_container(PairContainer *pc) {
+  contribs_.push_back(pc);
+  default_range_.push_back((int)default_range_.size());
```

Comment:
Avoid C-style casts in C++ code. `static_cast<int>` would be more typical C++ usage.

Reply:
Done